v6_carrier_py/master.py

```diff
@@ -105,12 +112,53 @@ def correlation_matrix(client: ClientContainerProtocol, data, keys=None, *args,
     If no keys are specified the datasets are joined on all columns with the same name.
     TODO: What if different datasets use different keys to mean the same thing? How do we specify this?
     """
-    results = _dispatch_tasks(client, data, *args, method='get_data', **kwargs)
+    combined_df = _combine_all_node_data(client, data, keys, *args, **kwargs)
+
+    return combined_df.corr()
+
+
+def train_model(client: ClientContainerProtocol, data, pipeline: Pipeline, features: List[str], target: str, keys=None,
```

Comment:
By allowing the user to add a Pipeline, the user can basically inject anything he wants. This is fine as long as you only return the metrics and you specify what kind of metrics. But it becomes tricky if you want to return the model. In that case you could quite easily steal data by executing a pipeline that .... well, steals data.

Reply:
Yeah I'm aware of that. This is basically what Chang had implemented. I see it as a starting. With pythons duck typing you can basically package any type of malicious code as a "data analysis pipeline".

Still, I think it is good if this function is generic for now. Maybe later it will be wrapped by another algorithm that constructs a pipeline server-side based on user input.